yjit/src/codegen.rs

```diff
@@ -5395,11 +5395,18 @@ fn jit_rb_str_getbyte(
     extern "C" {
         fn rb_str_getbyte(str: VALUE, index: VALUE) -> VALUE;
     }
-    // Raises when non-integers are passed in
-    jit_prepare_non_leaf_call(jit, asm);
 
     let index = asm.stack_opnd(0);
     let recv = asm.stack_opnd(1);
+
+    let arg0_type = asm.ctx.get_opnd_type(index.into());
+
+    // rb_str_getbyte should be leaf if the index is a fixnum
+    if arg0_type != Type::Fixnum {
+        // Raises when non-integers are passed in
+        jit_prepare_non_leaf_call(jit, asm);
+    }
```

Comment:
The name "arg0" seems kind of cryptic. I think this is enough?

```suggestion
    // rb_str_getbyte should be leaf if the index is a fixnum
    if asm.ctx.get_opnd_type(index.into()) != Type::Fixnum {
        // Raises when non-integers are passed in
        jit_prepare_non_leaf_call(jit, asm);
    }
```

Reply:
ya, seems fine to me, I was just copy / pasting from somewhere else 😅